src/task.cpp

```diff
@@ -0,0 +1,93 @@
+//===--- task.cpp - Task struct and [un]serializer ------------------------===//
+//
+// Copyright (c) 2023 Yuri Istomin
+//
+// Part of the NekoCdr's "tasknow" project
+// Licensed under the Apache License, Version 2.0 (the "License").
+// See http://www.apache.org/licenses/LICENSE-2.0 for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "task.h"
+
+#include "buffer.h"
+#include "defines.h"
+
+#include <cstddef>
+#include <cstdlib>
+#include <cstring>
+#include <memory>
+#include <utility>
+
+namespace tasknow {
+
+auto Task::operator==(const Task& right) const -> bool
+{
+    bool result{true};
+
+    result &= this->title == right.title;
+
+    return result;
+}
+
+auto Task::operator!=(const Task& right) const -> bool
+{
+    return !(*this == right);
+}
+
+auto serialize(Task* input) -> Buffer
+{
+    Buffer output{};
+    std::size_t title_size = input->title.size();
+
+    std::size_t raw_data_size{SizeofST + title_size};
+    std::size_t offsets[2]{
+        0,
+        SizeofST,
+    };
+
+    auto buff{std::make_unique<unsigned char[]>(raw_data_size)};
+
+    // NOLINTBEGIN(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wunsafe-buffer-usage"
```

Comment:
https://github.com/llvm/llvm-project/issues/64646#issuecomment-1747952699 suggests that this diagnostic is not going to be helpful to you whatsoever. I think you should disable it for the whole project in CMake, leaving a link to the issue as a comment nearby.

Reply:
Thx for link. Did it.